lib/flagsmith/engine/evaluation/mappers.rb

```diff
@@ -0,0 +1,233 @@
+        def self.map_identity_model_to_identity_context(identity, override_traits = nil)
+          # Use override traits if provided, otherwise use identity's traits
+          traits = override_traits || identity.identity_traits
+
+          # Map traits to a hash with trait key => trait value
+          traits_hash = {}
+          traits.each do |trait|
+            traits_hash[trait.trait_key] = trait.trait_value
+          end
+
+          {
+            identifier: identity.identifier,
+            key: identity.django_id&.to_s || identity.composite_key,
+            traits: traits_hash
+          }
+        end
+  
+        # Maps segment rule model to rule hash
+        #
+        # @param rule [Flagsmith::Engine::Segments::Rule] The segment rule model
+        # @return [Hash] Mapped rule with :type, :conditions, and :rules
+        def self.map_segment_rule_model_to_rule(rule)
+          result = {
+            type: rule.type
+          }
+
+          # Map conditions if present
+          if rule.conditions&.any?
+            result[:conditions] = rule.conditions.map do |condition|
+              {
+                property: condition.property,
+                operator: condition.operator,
+                value: condition.value
+              }
+            end
+          else
+            result[:conditions] = []
+          end
+
+          if rule.rules&.any?
+            result[:rules] = rule.rules.map { |nested_rule| map_segment_rule_model_to_rule(nested_rule) }
+          else
+            result[:rules] = []
+          end
+
+          result
+        end
+  
+        # Maps identity overrides to segments
+        #
+        # @param identity_overrides [Array<Flagsmith::Engine::Identity>] Array of identity override models
+        # @return [Hash] Segments hash for identity overrides
+        def self.map_identity_overrides_to_segments(identity_overrides)
+          require 'digest'
+
+          segments = {}
+          features_to_identifiers = {}
+
+          identity_overrides.each do |identity|
+            next if identity.identity_features.nil? || !identity.identity_features.any?
+
+            # Sort features by name for consistent hashing
+            sorted_features = identity.identity_features.to_a.sort_by { |fs| fs.feature.name }
+
+            # Create override keys for hashing
+            overrides_key = sorted_features.map do |fs|
+              {
+                feature_key: fs.feature.id.to_s,
+                name: fs.feature.name,
+                enabled: fs.enabled,
+                value: fs.get_value,
+                priority: WEAKEST_PRIORITY,
+                metadata: {
+                  flagsmithId: fs.feature.id
+                }
+              }
+            end
+
+            # Create hash of the overrides to group identities with same overrides
+            overrides_hash = Digest::SHA1.hexdigest(overrides_key.to_json)
```

Comment:
Consider one of the following alternatives.


1. Using `inspect` instead of `to_json` might be more performant, and doesn't require us to avoid `Inf`:
```suggestion
            overrides_hash = Digest::SHA1.hexdigest(overrides_key.inspect)
```

2. I'm not too familiar with Ruby's [`Object::hash`](https://apidock.com/ruby/Object/hash), but it could be sufficient in our case. We could drop the `digest` dependency that way:

```suggestion
            overrides_hash = overrides_key.hash
```